readers/src/main/scala/com/salesforce/op/readers/ParquetProductReader.scala

```diff
@@ -56,3 +56,33 @@ class ParquetProductReader[T <: Product : Encoder]
     maybeRepartition(data, params)
   }
 }
+
+/**
+  * Data Reader for Parquet events, where there may be multiple records for a given key. Each parquet record
+  * will be automatically converted to type T that defines an [[Encoder]].
+  * @param readPath default path to data
+  * @param key      function for extracting key from record
+  * @param aggregateParams params for time-based aggregation
+  * @tparam T
+  */
+class AggregateParquetProductReader[T <: Product : Encoder : WeakTypeTag]
+(
+  readPath: Option[String],
+  key: T => String,
+  val aggregateParams: AggregateParams[T]
+)extends ParquetProductReader[T](readPath, key) with AggregateDataReader[T]
+
+/**
+  * Data Reader for Parquet events, when computing conditional probabilities. There may be multiple records for
+  * a given key. Each parquet record will be automatically converted to type T that defines an [[Encoder]].
+  * @param readPath default path to data
+  * @param key      function for extracting key from record
+  * @param conditionalParams params for conditional aggregation
+  * @tparam T
+  */
+class ConditionalParquetProductReader[T <: Product : Encoder : WeakTypeTag]
+(
+  readPath: Option[String],
+  key: T => String,
+  val conditionalParams: ConditionalParams[T]
+)extends ParquetProductReader[T](readPath, key) with ConditionalDataReader[T]
```

Comment:
please fix scalastyle warnings

Reply:
@tovbinm Seems like we have are going to have [similar logic](https://github.com/salesforce/TransmogrifAI/blob/master/readers/src/test/scala/com/salesforce/op/readers/CSVProductReadersTest.scala#L119-L163) for testing these readers. Will look at the possibility of extracting these as Table Tests and update the PR.

Reply:
Any tests included? Why not?